src/main/java/au/id/micolous/metrodroid/transit/kmt/KMTUtil.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * KMTUtil.java
+ *
+ * Authors:
+ * Bondan Sumbodo <sybond@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package au.id.micolous.metrodroid.transit.kmt;
+
+import net.kazzz.felica.lib.Util;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+final class KMTUtil {
+    private KMTUtil() {
+    }
+
+    static Calendar extractDate(byte[] data) {
+        int fulloffset = Util.toInt(data[0], data[1], data[2], data[3]);
+        if (fulloffset == 0) {
+            return null;
+        }
+        Calendar c = new GregorianCalendar(KMTTransitData.TIME_ZONE);
+        c.setTime(new Date((long) (fulloffset + 946684758) * 1000));
```

Comment:
Please explicitly define epochs as a constant and as a date, as it makes it easier to read.  eg:

https://github.com/micolous/metrodroid/blob/9b8550615f8bf517fd5f7bd63ff188d7bf08ac26/src/main/java/au/id/micolous/metrodroid/transit/troika/TroikaTransitData.java#L58-L65

I'm not sure that the values you have here are exactly correct:

```
$ TZ=Asia/Jakarta date --date='@946684758'
Sat Jan  1 06:59:18 WIB 2000
$ TZ=UTC date --date='@946684758'
Fri Dec 31 23:59:18 UTC 1999
```

I think you're trying to set the epoch as 2000-01-01 00:00 UTC?

`ClipperTransitData` has a more complex example where the epoch is in UTC, but the rest of the card has dates in local time.

Reply:
Thanks for pointing this out. Actually I'm not aware that epoch value are ` 2000-01-01 00:00 UTC`. I just manually calc real time during gate-out and see the diff in the data. I will check whether this epoch value is correct or not with the data.

Reply:
After playing around with some data, it seems spoch is not `2000-01-01 00:00 UTC`. Anyway following code will work, and easier to read :D

    static {
         GregorianCalendar epoch = new GregorianCalendar(TIME_ZONE);
         epoch.set(2000, 0, 1, 7, 0, 0);
         KMT_EPOCH = epoch.getTimeInMillis();
     }

Here is sample data: `22 F1 A7 E6`, and should be decoded as `30 Jul 2018 17:41 WIB`.